### PR TITLE
Report all applied option names even if cost is zero

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

A single upgrade can include multiple options, e.g.:
```
  - upgrade_name: Lighting
    options:
      - option: Lighting|100% LED
        costs:
          - value: 0.75
            multiplier: Floor Area, Lighting (ft^2)
        lifetime: 25
      - option: Lighting Interior Use|100% Usage
      - option: Lighting Other Use|100% Usage
      - option: Holiday Lighting|No Exterior Use
```
In the example above, costs are only assigned under 1 of the 4 options. Based on code in `UpgradeCosts`, this means options 2 - 4 do not get reported. This PR changes that behavior to report option names 2 - 4 even if their costs are zero.

Now reports:
upgrade_costs.option_01_cost_usd | upgrade_costs.option_01_lifetime_yrs | upgrade_costs.option_01_name | upgrade_costs.option_02_name | upgrade_costs.option_03_name | upgrade_costs.option_04_name
-- | -- | -- | -- | -- | --
853.5 | 25 | Lighting\|100% LED | Lighting Interior Use\|100% Usage | Lighting Other Use\|100% Usage | Holiday Lighting\|No Exterior Use

## Checklist

Not all may apply:

- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] Changelog has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected integration/regression test changes

### CODE REVIEW

File: measures/UpgradeCosts/measure.rb [hunk 1]
```diff
@@ -117,14 +117,16 @@ def run(runner, user_arguments)
...
-      runner.registerInfo("Registering (blank) for #{upgrade_cost_name}.")
+      register_value(runner, upgrade_cost_name, 0.0)
+      runner.registerInfo("Registering 0.0 for #{upgrade_cost_name}.")
       return true
     end
 
     # Obtain cost multiplier values and calculate upgrade costs
     upgrade_cost = 0.0
     option_cost_pairs.keys.each do |option_num|
+      next if option_cost_pairs[option_num].empty?
```
Comment (rajeee) [inline measures/UpgradeCosts/measure.rb:128]: Why was this necessary to add?

Comment (joseph-robertson) [reply measures/UpgradeCosts/measure.rb:128]: The loop iterates through all 25 options, but we don't always have all 25 populated.

**Review by rajeee — APPROVED** [review]
Looks good. Thank you.
